test/type-error.js

```diff
@@ -0,0 +1,563 @@
+		`if (!foo instanceof Foo) {
+			throw new TypeError('Expected Foo being bar!');
+		}`,
+		`if (foo instanceof Foo === false) {
+			throw new TypeError('Expected Foo being bar!');
+		}`,
+		`throw new Error('💣')`,
+		`if (!Number.isNaN(foo) && foo === 10) {
+			throw new Error('foo is not 10!');
+		}`,
+		`function foo(foo) {
+			if (!Number.isNaN(foo) && foo === 10) {
+				timesFooWas10 += 1;
+				if (calculateAnswerToLife() !== 42) {
+					openIssue('Your program is buggy!');
+				} else {
+					return printAwesomeAnswer(42);
+				}
+				throw new Error('foo is 10');
+			}
+		}`,
+		`function foo(foo) {
+			if (!Number.isNaN(foo)) {
+				timesFooWas10 += 1;
+				if (calculateAnswerToLife({with: foo}) !== 42) {
+					openIssue('Your program is buggy!');
+				} else {
+					return printAwesomeAnswer(42);
+				}
+				throw new Error('foo is 10');
+			}
+		}`,
+		`if (!x.isFudge()) {
+			throw new Error('x is no fudge!');
+		}`,
+		`if (!_.isFudge(x)) {
+			throw new Error('x is no fudge!');
+		}`
+	],
+	invalid: [
+		{
+			code: `if (Array.isArray(foo)) {
+				throw new Error('foo is an Array');
+			}`,
+			output: `if (Array.isArray(foo)) {
+				throw new TypeError('foo is an Array');
+			}`,
+			errors
+		},
+		{
+			code: `if (foo instanceof bar) {
+				throw new Error(foobar);
+			}`,
+			output: `if (foo instanceof bar) {
+				throw new TypeError(foobar);
+			}`,
+			errors
+		},
+		{
+			code: `if (_.isElement(foo)) {
+				throw new Error();
+			}`,
+			output: `if (_.isElement(foo)) {
+				throw new TypeError();
+			}`,
+			errors
+		},
+		{
+			code: `if (_.isElement(foo)) {
+				throw new Error;
+			}`,
+			output: `if (_.isElement(foo)) {
+				throw new TypeError;
+			}`,
+			errors
+		},
+		{
+			code: `if (wrapper._.isElement(foo)) {
+				throw new Error;
+			}`,
+			output: `if (wrapper._.isElement(foo)) {
+				throw new TypeError;
+			}`,
+			errors
+		},
+		{
+			code: `if (typeof foo == 'Foo' || 'Foo' === typeof foo) {
+				throw new Error();
+			}`,
+			output: `if (typeof foo == 'Foo' || 'Foo' === typeof foo) {
+				throw new TypeError();
+			}`,
+			errors
+		},
+		{
+			code: `if (Number.isFinite(foo) && Number.isSafeInteger(foo) && Number.isInteger(foo)) {
+				throw new Error();
+			}`,
+			output: `if (Number.isFinite(foo) && Number.isSafeInteger(foo) && Number.isInteger(foo)) {
+				throw new TypeError();
+			}`,
+			errors
+		},
+		{
+			code: `if (wrapper.n.isFinite(foo) && wrapper.n.isSafeInteger(foo) && wrapper.n.isInteger(foo)) {
+				throw new Error();
+			}`,
+			output: `if (wrapper.n.isFinite(foo) && wrapper.n.isSafeInteger(foo) && wrapper.n.isInteger(foo)) {
+				throw new TypeError();
+			}`,
+			errors
+		},
+		{
+			code: `if (wrapper.f.g.n.isFinite(foo) && wrapper.g.n.isSafeInteger(foo) && wrapper.n.isInteger(foo)) {
+				throw new Error();
+			}`,
+			output: `if (wrapper.f.g.n.isFinite(foo) && wrapper.g.n.isSafeInteger(foo) && wrapper.n.isInteger(foo)) {
+				throw new TypeError();
+			}`,
+			errors
```

Comment:
By the way, since you're writing pretty much the same boilerplate everywhere, you could make a test helper.

```js
function errorTestCase(code) {
  return {
    code,
    output: code.replace('new Error(', 'new TypeError('),
    errors
  };
}

errorTestCase(`if (wrapper.f.g.n.isFinite(foo) && wrapper.g.n.isSafeInteger(foo) && wrapper.n.isInteger(foo)) {
  throw new Error();
}`)
```